src/mkappimage/mkappimage.go

```diff
@@ -51,6 +51,15 @@ func bootstrapMkAppImage(c *cli.Context) error {
 		fileToAppImageOutput = c.Args().Get(1)
 	}
 
+    // make sure the output directory exists before continuing
+    outputDirectory := fileToAppImageOutput
+    if strings.HasSuffix(fileToAppImageOutput, ".AppImage") {
```

Comment:
`src/mkappimage/mkappimage.go:56:8: undefined: strings`

I also don't really understand this `if` statement: Do you expect the output given to be either a directory or a file, and you assume it is a file if the path ends with `.AppImage`? That's rather weird.

My preference would be to allow the destination to only be a path to the destination file, then you could use the file API for all parts: (pseudocode) `assert(fileToAppImageOutput.exists() && fileToAppImageOutput.isFile() && fileToAppImageOutput.parent().exists())`

Reply:
> `src/mkappimage/mkappimage.go:56:8: undefined: strings`

Thanks. good catch.

> I also don't really understand this if statement: Do you expect the output given to be either a directory or a file, and you assume it is a file if the path ends with .AppImage? That's rather weird.

It might be. I don't have a better solution as of now.

> My preference would be to allow the destination to only be a path to the destination file, then you could use the file API for all parts: (pseudocode) assert(fileToAppImageOutput.exists() && fileToAppImageOutput.isFile() && fileToAppImageOutput.parent().exists())

Thanks, unfortunately, that doesn't help. The destination is something which would be created in future. `fileToAppImageOutput.exists()` will always give `False`, because that's something which `mkappimage` is going to create, right? So, the pseudocode will not help. 

What API would you suggest me to check if a file which is going to be written in future is a `file` or a `directory`?

Reply:
> src/mkappimage/mkappimage.go:56:8: undefined: strings

fixed in https://github.com/probonopd/go-appimage/pull/104/commits/f41af3fa003c98c8f0cb6e3373b01ec24d6fbbc1

Reply:
Oh, sorry, I got things the wrong way. Let me try again: `assert(!fileToAppImageOutput.exists() && fileToAppImageOutput.parent().exists() && fileToAppImageOutput.parent().isDirectory())`

> What API would you suggest me to check if a file which is going to be written in future is a file or a directory?

You know that you won't create a directory, so you simply need to check that the path does not exist yet. And for the parent, only thing you need to know is that it exists and that it is a directory (= not a file) so that you can write in it.

If the overwrite flag is set, `!fileToAppImageOutput.exists()` becomes `fileToAppImageOutput.exists() && fileToAppImageOutput.isFile()`.

Reply:
> assert(!fileToAppImageOutput.exists() && fileToAppImageOutput.parent().exists() && fileToAppImageOutput.parent().isDirectory())

Agreed, that looks better. As of now, I already check if the parent directory exists before creating the appimage. Now I need to check if the parent is a directory. Thanks for your suggestion. That would do. Let me push some more commits.

Reply:
Yea, should do now. I have removed the `--overwrite` flag. Now the ambiguity should vanish :smile: 

Thanks for the review once again